metis-media-service/src/main/java/eu/europeana/metis/mediaservice/MediaProcessor.java

```diff
@@ -0,0 +1,441 @@
+package eu.europeana.metis.mediaservice;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pdfbox.cos.COSName;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDResources;
+import org.apache.pdfbox.pdmodel.common.PDMetadata;
+import org.apache.pdfbox.pdmodel.graphics.PDXObject;
+import org.apache.pdfbox.pdmodel.graphics.image.PDImageXObject;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.apache.tika.Tika;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MediaProcessor implements Closeable {
+	
+	private static final Logger log = LoggerFactory.getLogger(MediaProcessor.class);
+	
+	private static final int[] THUMB_SIZE = { 200, 400 };
+	private static final String[] THUMB_SUFFIX = { "-MEDIUM", "-LARGE" };
+	
+	private static File tempDir;
+	private static File colormapFile;
+	private static String magickCmd;
+	private static String ffprobeCmd;
+	private static Tika tika;
+	
+	private ExecutorService commandIOThreadPool = Executors.newFixedThreadPool(2);
```

Comment:
Number 2 can be a constant field.

Reply:
I'd prefer to keep the list of fields clean. Defining that 2 is the THREAD_COUNT would not make anything clearer.